tidy3d/components/data/sim_data.py

```diff
@@ -779,3 +640,162 @@ def plot_scalar_array(
         ax.set_ylim(min(y_coord_values), max(y_coord_values))
 
         return ax
+
+
+class SimulationData(AbstractYeeGridSimulationData):
```

Comment:
Nice, interesting, I agree. It's kind of creating a separation between the solvers whilst still sharing core aspects to some level. I"m wondering how this works in the context of EME specific extensions? Would you edit AbstractYeeGridSimulationData directly, and do these extensions have to couple in to FDTDSimulationData inherently/will they be supported?